src/modules/UI/components/ErrorAlert/actions.js

```diff
@@ -1,13 +1,25 @@
 // @flow
+import { Alert } from 'react-native'
+
+import * as actions from '../../../../actions/indexActions.js'
+import s from '../../../../locales/strings.js'
+import type { Dispatch } from '../../../../modules/ReduxTypes'
 
 const PREFIX = 'UI/components/ErrorAlert/'
 export const DISPLAY_ERROR_ALERT = PREFIX + 'DISPLAY_ERROR_ALERT'
 export const DISMISS_ERROR_ALERT = PREFIX + 'DISMISS_ERROR_ALERT'
 
-export const displayErrorAlert = (message: string) => ({
+export const displayErrorAlertStore = (message: string) => ({
   type: DISPLAY_ERROR_ALERT,
   data: { message }
 })
+export const displayErrorAlert = (message: string) => (dispatch: Dispatch) => {
+  if (message === 'Invalid OTP token') {
+    Alert.alert(s.strings.otp_out_of_sync_title, s.strings.otp_out_of_sync_body)
+    return
+  }
+  dispatch(actions.dispatchActionObject(DISPLAY_ERROR_ALERT, { message }))
```

Comment:
what is the difference betrween 
```js
dispatch(actions.dispatchActionObject(DISPLAY_ERROR_ALERT, { message }))
```
and 
```js
dispatch(displayErrorAlert(message))
```
?

Reply:
The difference between the two is 
calling `dispatch(displayErrorAlert(message))` will be recursive loop since you are in a thunk this is that function, `dispatch(actions.dispatchActionObject(DISPLAY_ERROR_ALERT, {message}))` does what the old function did, it dispatches a regular action creator with the correct type `DISPLAY_ERROR_ALERT` and an object with a property `message`...